source/beamline/AMTemperatureMonitor.cpp

```diff
@@ -0,0 +1,102 @@
+#include "AMTemperatureMonitor.h"
+
+AMTemperatureMonitor::AMTemperatureMonitor(const QString &name, const QString &units, QObject *parent) :
+	AMConnectedControl(name, units, parent)
+{
+	goodStatusValue_ = -1;
+	badStatusValue_ = -1;
+
+	statusControl_ = 0;
+	valueControl_ = 0;
+}
+
+AMTemperatureMonitor::~AMTemperatureMonitor()
+{
+
+}
+
+bool AMTemperatureMonitor::canMeasure() const
+{
+	return (valueControl_ && valueControl_->canMeasure());
+}
+
+bool AMTemperatureMonitor::isGood() const
+{
+	bool result = false;
+
+	if (statusControl_ && statusControl_->canMeasure())
+		result = (statusControl_->value() == goodStatusValue_);
+
+	return result;
+}
+
+bool AMTemperatureMonitor::isBad() const
+{
+	bool result = false;
+
+	if (statusControl_ && statusControl_->canMeasure())
+		result = (statusControl_->value() == badStatusValue_);
+
+	return result;
+}
```

Comment:
just pop up a question: is there a state of `in-between`(is not `good` and  not `bad`?